cmd/controller/main.go

```diff
@@ -44,17 +46,33 @@ const (
 )
 
 func main() {
+	var logLevel string
+
 	flag.IntVar(&ctrlConcurrency, "concurrency", 10, "Max concurrent reconciles")
 	flag.StringVar(&ctrlNamespace, "namespace", "", "Namespace to watch")
 	flag.BoolVar(&allowSharedServiceAccount, "dangerous-allow-shared-service-account",
 		false, "If set to true, allow use of shared service account instead of per-app service accounts")
 	flag.BoolVar(&enablePprof, "dangerous-enable-pprof", false, "If set to true, enable pprof on "+pprofListenAddr)
+	flag.StringVar(&logLevel, "loglevel", zapcore.InfoLevel.String(), "Set log level from one of the available options: debug/info/warn/error/dpanic/panic/fatal")
 	flag.Parse()
 
-	logf.SetLogger(zap.Logger(false))
+	level := uberzap.NewAtomicLevel()
+	logLevelErr := level.UnmarshalText([]byte(logLevel))
+	if logLevelErr != nil {
+		level.SetLevel(zapcore.InfoLevel)
+	}
+	logf.SetLogger(zap.New(func(o *zap.Options) {
+		o.Level = &level
+		o.Development = false
+	}))
+
 	entryLog := log.WithName("entrypoint")
 	entryLog.Info("kapp-controller", "version", Version)
 
+	if logLevelErr != nil {
+		entryLog.Error(logLevelErr, "unable to unmarshal log level", "loglevel", logLevel)
```

Comment:
we should probably exit here as well `os.Exit(1)` since we've received invalid configuration.

Reply:
Done with our last change.